packages/experimental/src/user-interaction-instrumentation.ts

```diff
@@ -0,0 +1,206 @@
+const SPAN_KEY = '__HNY_SPAN';
+
+const DEFAULT_EVENT_NAMES: EventName[] = ['click'];
+
+type EventName = keyof GlobalEventHandlersEventMap;
+
+type Listener = {
+  eventName: EventName;
+  handler: (event: Event) => void;
+};
+
+interface UserInteractionInstrumentationConfig extends InstrumentationConfig {
+  eventNames?: EventName[];
+  rootNodeId?: string;
+}
+
+export class UserInteractionInstrumentation extends InstrumentationBase {
+  protected _config: UserInteractionInstrumentationConfig;
+  private _is_enabled: boolean;
+  private _listeners: Listener[];
+
+  constructor(config: UserInteractionInstrumentationConfig = {}) {
+    super(INSTRUMENTATION_NAME, VERSION, config);
+    this._config = config;
+    this._is_enabled = this._config.enabled ?? false;
+
+    // enable() gets called by our superclass constructor
+    // @ts-expect-error this may get set in enable()
+    this._listeners = this._listeners ?? [];
+  }
+
+  protected init() {}
+
+  public enable() {
+    if (this._is_enabled) {
+      return;
+    }
+    const rootNode = this.getRootNode();
+
+    // enable() gets called by our superclass constructor
+    // meaning our private fields aren't initialized yet!!
+    this._listeners = [];
+    //
+    const eventNames = this._config.eventNames ?? DEFAULT_EVENT_NAMES;
+    eventNames.forEach((eventName) => {
+      // we need a stable reference to this handler so that we can remove it later
+      const handler = createGlobalEventListener(
+        eventName,
+        this._config.rootNodeId,
+        () => this._is_enabled,
+      );
+      this._listeners.push({ eventName, handler });
+
+      // capture phase listener to kick in before any other listeners
+      rootNode.addEventListener(eventName, handler, { capture: true });
+
+      // bubble phase listener gets called at the end, if user space doesn't call e.stopPropagation()
+      rootNode.addEventListener(eventName, endSpan);
+    });
+
+    this._is_enabled = true;
+  }
+
+  private getRootNode() {
+    if (this._config.rootNodeId) {
+      const rootNode = document.getElementById(this._config.rootNodeId);
+      if (rootNode === null) {
+        this._diag.warn(`Root Node id: ${this._config.rootNodeId} not found!`);
+        return document;
+      }
+      return rootNode;
+    }
+    return document;
+  }
+
+  public disable(): void {
+    this._is_enabled = false;
+    this._listeners.forEach(({ eventName, handler }) => {
+      document.removeEventListener(eventName, handler, { capture: true });
+      document.removeEventListener(eventName, endSpan);
+    });
+    this._listeners = [];
+  }
+}
+
+const shouldCreateSpan = (
+  event: Event,
+  element: EventTarget | null,
+  eventName: EventName,
+  rootNodeId: string | undefined,
+): element is HTMLElement => {
+  // @ts-expect-error we set this field in the handler, if it's already here then we've already seen this event
+  if (event[SPAN_KEY]) {
+    return false;
+  }
```

Comment:
I'd love to see this as a WeakMap or something so we can avoid modifying the event attributes, but this works for now!